test/suites/integration/bsp/challenge-cycle.test.ts

```diff
@@ -1,78 +1,172 @@
+        await userApi.call.proofsDealerApi.getChallengePeriod(DUMMY_BSP_ID);
+      assert(challengePeriodResult.isOk);
+      const challengePeriod = challengePeriodResult.asOk.toNumber();
+      // We get the challenge ticks tolerance.
+      const challengeTicksTolerance = Number(userApi.consts.proofsDealer.challengeTicksTolerance);
+      // And finally we calculate the next deadline tick.
+      const nextDeadlineTick =
+        lastTickBspSubmittedProof + challengePeriod + challengeTicksTolerance;
+
+      // Advance to BSP deadline.
+      const currentBlock = await userApi.rpc.chain.getBlock();
+      const currentBlockNumber = currentBlock.block.header.number.toNumber();
+      const blocksToAdvance = nextDeadlineTick - currentBlockNumber;
+      let blockResult: SealedBlock | undefined;
+      for (let i = 0; i < blocksToAdvance; i++) {
+        blockResult = await userApi.sealBlock();
+      }
+
+      // Check for event of slashable BSP.
+      userApi.assertEvent("proofsDealer", "SlashableProvider", blockResult?.events);
+    });
+
+    it(
+      "BSP resumes and sends pending proofs",
+      {
+        skip: "Sending pending proofs is not yet implemented."
+      },
+      async () => {}
+    );
+
+    it(
+      "BSP is challenged again and correctly submits proof",
+      {
+        skip: "Correctly resuming BSP is not yet implemented."
+      },
+      async () => {
+        // Resume BSP.
+        await resumeBspContainer(NODE_INFOS.bsp.containerName);
+
+        // Advance to the next tick the BSP should submit a proof for, that is after the current block.
+        // We first get the last tick for which the BSP submitted a proof.
+        const lastTickResult =
+          await bspApi.call.proofsDealerApi.getLastTickProviderSubmittedProof(DUMMY_BSP_ID);
+        assert(lastTickResult.isOk);
+        const lastTickBspSubmittedProof = lastTickResult.asOk.toNumber();
+        // Then we get the challenge period for the BSP.
+        const challengePeriodResult =
+          await bspApi.call.proofsDealerApi.getChallengePeriod(DUMMY_BSP_ID);
+        assert(challengePeriodResult.isOk);
+        const challengePeriod = challengePeriodResult.asOk.toNumber();
+        // Then we calculate the next challenge tick.
+        let nextChallengeTick = lastTickBspSubmittedProof + challengePeriod;
+        // Increment challenge periods until we get a number that is greater than the current tick.
+        const currentTick = (await bspApi.call.proofsDealerApi.getCurrentTick()).toNumber();
+        while (currentTick >= nextChallengeTick) {
+          // Go one challenge period forward.
+          nextChallengeTick += challengePeriod;
+        }
+        // Finally, advance to the next challenge tick.
+        const currentBlock = await userApi.rpc.chain.getBlock();
+        const currentBlockNumber = currentBlock.block.header.number.toNumber();
+        const blocksToAdvance = nextChallengeTick - currentBlockNumber;
```

Comment:
Your assumption is correct, but I don't think we should add such parameter because `get_next_deadline_tick` should return the **deadline**, which includes the tolerance. We could have another runtime API for getting the next challenge block, but we already have runtime APIs for the last tick a provider submitted a proof for, and the challenge period of a provider. You can get the next challenge block by adding those two. And as you can see from this PR, adding a runtime API takes A LOT of boilerplate code.

Reply:
Ya I assumed that was the reason, but I like the way we offload StorageHub protocols to runtime APIs, since it simplifies client side development. If you don’t think it’s that much of a gain, it’s fine by me, but I would have done it.

Reply:
Yeah, feel free to add it whenever if you want.